DataFormats/EgammaCandidates/interface/GsfElectron.h

```diff
@@ -355,19 +356,24 @@ namespace reco {
 
   public:
     struct ShowerShape {
-      float sigmaEtaEta;         // weighted cluster rms along eta and inside 5x5 (absolute eta)
-      float sigmaIetaIeta;       // weighted cluster rms along eta and inside 5x5 (Xtal eta)
-      float sigmaIphiIphi;       // weighted cluster rms along phi and inside 5x5 (Xtal phi)
-      float e1x5;                // energy inside 1x5 in etaxphi around the seed Xtal
-      float e2x5Max;             // energy inside 2x5 in etaxphi around the seed Xtal (max bwt the 2 possible sums)
-      float e5x5;                // energy inside 5x5 in etaxphi around the seed Xtal
-      float r9;                  // ratio of the 3x3 energy and supercluster energy
-      float hcalDepth1OverEcal;  // hcal over ecal seed cluster energy using 1st hcal depth (using hcal towers within a cone)
-      float hcalDepth2OverEcal;  // hcal over ecal seed cluster energy using 2nd hcal depth (using hcal towers within a cone)
-      std::vector<CaloTowerDetId> hcalTowersBehindClusters;  //
-      float hcalDepth1OverEcalBc;  // hcal over ecal seed cluster energy using 1st hcal depth (using hcal towers behind clusters)
-      float hcalDepth2OverEcalBc;  // hcal over ecal seed cluster energy using 2nd hcal depth (using hcal towers behind clusters)
+      float sigmaEtaEta;    // weighted cluster rms along eta and inside 5x5 (absolute eta)
+      float sigmaIetaIeta;  // weighted cluster rms along eta and inside 5x5 (Xtal eta)
+      float sigmaIphiIphi;  // weighted cluster rms along phi and inside 5x5 (Xtal phi)
+      float e1x5;           // energy inside 1x5 in etaxphi around the seed Xtal
+      float e2x5Max;        // energy inside 2x5 in etaxphi around the seed Xtal (max bwt the 2 possible sums)
+      float e5x5;           // energy inside 5x5 in etaxphi around the seed Xtal
+      float r9;             // ratio of the 3x3 energy and supercluster energy
+      float hcalDepth1OverEcal;  // run2 hcal over ecal seed cluster energy using 1st hcal depth (using hcal towers within a cone)
+      float hcalDepth2OverEcal;  // run2 hcal over ecal seed cluster energy using 2nd hcal depth (using hcal towers within a cone)
+      float hcalDepth1OverEcalBc;  // run2 hcal over ecal seed cluster energy using 1st hcal depth (using hcal towers behind clusters)
+      float hcalDepth2OverEcalBc;  // run2 hcal over ecal seed cluster energy using 2nd hcal depth (using hcal towers behind clusters)
+      std::array<float, 7>
+          hcalOverEcal;  // run3 hcal over ecal seed cluster energy per depth (using rechits within a cone)
+      std::array<float, 7>
+          hcalOverEcalBc;  // run3 hcal over ecal seed cluster energy per depth (using rechits behind clusters)
+      std::vector<CaloTowerDetId> hcalTowersBehindClusters;
       bool invalidHcal;  // set to true if the hcal energy estimate is not valid (e.g. the corresponding tower was off or masked)
+      bool pre7DepthHcal;
```

Comment:
I think a one-line comment to describe why this is needed can be helpful for future readers, e.g.
```C++
bool pre7DepthHcal; //to work around an ioread rule issue on legacy RECO files
```